File: Programmers/12977.py
def solution(nums):
    def prime(n):
        for i in range(2, int(n/2)):
            if n % i == 0:
                return False
        return True
    
    def dfs(idx, cnt, m):
        if m == 3:
            if prime(cnt):
                return 1
            else:
                return 0
        if idx == len(nums):
            return 0
        return dfs(idx+1, cnt+nums[idx], m+1) + dfs(idx+1, cnt, m)
    return dfs(0, 0, 0)

File: Programmers/12977.cpp
#include <vector>
#include <iostream>
using namespace std;
bool check(int n){      // 2*4 = 4*2 임을 확인하여 반만 검사 
    for(int i=2;i<n/2;i++)
        if(n%i==0)
            return false;
    return true;
}
int solution(vector<int> nums) {
    int answer = 0;
    int N=nums.size();
    for(int i=0;i<N;i++)
        for(int j=i+1;j<N;j++)
            for(int k=j+1;k<N;k++){
                if(check(nums[i]+nums[j]+nums[k]))
                    answer++;
            }
    return answer;
}
